src/Symfony/Component/Cache/Traits/PhpFilesTrait.php

```diff
@@ -32,6 +32,35 @@ public static function isSupported()
     }
 
     /**
+     * @return bool
+     */
+    public function prune()
+    {
+        $time = time();
+        $pruned = true;
+        $allowCompile = static::opcacheAllowCompile();
+
+        set_error_handler($this->includeHandler);
+        try {
+            foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->directory, \FilesystemIterator::SKIP_DOTS), \RecursiveIteratorIterator::LEAVES_ONLY, \RecursiveIteratorIterator::CATCH_GET_CHILD) as $file) {
```

Comment:
The `CATCH_GET_CHILD` flag causes iterator children calls to not throw an exception (for example if an unreadable directory is encountered) which made sense to me in the context of `prune`. Do you think it's needed here? Is it needed elsewhere? Or should be removed?